### DISCUSSION TITLE
Model attribute unsets/disappears if validation rule not present

### DISCUSSION THREAD

**Discussion by ridxery** [livewire/livewire#3822] [7. Bugs (archived)] [answered]
Lets say you have 

public Invoice $invoice;
public $foo;

---

Invoice has 
['total_amount', 'total_tax', 'status', 'currency_id' ...]

assume you dont have validation for 'invoice.total_amount'  and you unset/null $foo then 'invoice.total_amount' just disappears/unsets. I solved this just adding validation rule for 'invoice.total_amount' and everything worked it persisted. 

FYI:  
+) livewire version: 2.6.5
+) $invoice->total_amount is dynamically/server-side calculated value (no wire:model on any html input)
+) I rolled back to my latest stable commit and only add  validation rule (1 line of code ['invoice.total_amount' => 'required|numeric|min:0]) and it persisted.

**Comment by joshhanley** [comment, marked as answer]
@ridxery yeah that is intentional for security reasons. Any properties you want to persist when using model binding, need to have a rule defined (even if it's just `'nullable'`).

See in the docs here https://laravel-livewire.com/docs/2.x/properties#binding-models

> Note: For this to work, you have a validation entry in the `$rules` property for any model attributes you want to bind to. Otherwise, an error will be thrown.

Although you are not binding to it, the only way Livewire keeps state is by sending the data to the front end, and the rules array is a whitelist for what data to send.

Hope this helps!

**Reply by ridxery** [reply]
Thanks Josh, my bad not noticed it on docs.

**Reply by ighmouraceneb** [reply]
This was not the case in versions prior to 2.6. 
#3830

**Reply by joshhanley** [reply]
@ighmouraceneb actually that was still the case in 2.5 (and has been since model binding was introduced in V2), you still need rules to get data to the front end, when using model binding for security purposes. Your issue is a seperate issue to this one, and is to do with how dirty data is being merged back into original object, not a rules problem (as your example has rules in it).

**Reply by ighmouraceneb** [reply]
Thank you Joh.
